packages/react-dom/src/__tests__/ReactFlightIntegration-test.js

```diff
@@ -0,0 +1,333 @@
+    const [GamesModel, , rejectGamesModel] = makeDelayedText();
+    function ProfileMore() {
+      return {
+        avatar: <Text>:avatar:</Text>,
+        friends: <FriendsModel>:friends:</FriendsModel>,
+        posts: <PostsModel>:posts:</PostsModel>,
+        games: <GamesModel>:games:</GamesModel>,
+      };
+    }
+    function ProfileModel() {
+      return {
+        photos: <PhotosModel>:photos:</PhotosModel>,
+        name: <NameModel>:name:</NameModel>,
+        more: <ProfileMore />,
+      };
+    }
+
+    // View
+    function ProfileDetails({result}) {
+      return (
+        <div>
+          {result.model.name}
+          {result.model.more.avatar}
+        </div>
+      );
+    }
+    function ProfileSidebar({result}) {
+      return (
+        <div>
+          {result.model.photos}
+          {result.model.more.friends}
+        </div>
+      );
+    }
+    function ProfilePosts({result}) {
+      return <div>{result.model.more.posts}</div>;
+    }
+    function ProfileGames({result}) {
+      return <div>{result.model.more.games}</div>;
+    }
+    function ProfilePage({result}) {
+      return (
+        <>
+          <Suspense fallback={<p>(loading)</p>}>
+            <ProfileDetails result={result} />
+            <Suspense fallback={<p>(loading sidebar)</p>}>
+              <ProfileSidebar result={result} />
+            </Suspense>
+            <Suspense fallback={<p>(loading posts)</p>}>
+              <ProfilePosts result={result} />
+            </Suspense>
+            <ErrorBoundary fallback={e => <p>{e.message}</p>}>
+              <Suspense fallback={<p>(loading games)</p>}>
+                <ProfileGames result={result} />
+              </Suspense>
+            </ErrorBoundary>
+          </Suspense>
+        </>
+      );
+    }
+
+    let {writable, readable} = getTestStream();
+    ReactFlightDOMServer.pipeToNodeWritable(<ProfileModel />, writable);
+    let result = ReactFlightDOMClient.readFromReadableStream(readable);
+
+    let container = document.createElement('div');
+    ReactDOM.render(<ProfilePage result={result} />, container);
+    expect(container.innerHTML).toBe(
+      '<p style="display: none;">(loading sidebar)</p>' +
+        '<p style="display: none;">(loading posts)</p>' +
+        '<p style="display: none;">(loading games)</p>' +
+        '<p>(loading)</p>',
+    );
+
+    // This isn't enough to show anything.
+    resolveFriendsModel();
+    await act(async () => {
+      jest.runAllTimers();
+    });
+    expect(container.innerHTML).toBe(
+      '<p style="display: none;">(loading sidebar)</p>' +
+        '<p style="display: none;">(loading posts)</p>' +
+        '<p style="display: none;">(loading games)</p>' +
+        '<p>(loading)</p>',
+    );
+
+    // We can now show the details. Sidebar and posts are still loading.
+    resolveNameModel();
+    await act(async () => {
+      jest.runAllTimers();
+    });
+    await act(async () => {
+      jest.runAllTimers();
```

Comment:
This is again two of them. This shouldn't be needed since act should be doing it recursively.